x-pack/plugins/lens/public/xy_visualization/get_legend_action.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { LegendAction, XYChartSeriesIdentifier } from '@elastic/charts';
+import type { LayerArgs } from './types';
+import type { LensMultiTable, LensFilterEvent } from '../types';
+import { LegendActionPopover } from '../shared_components';
+
+export const getLegendAction = (
+  filteredLayers: LayerArgs[],
+  tables: LensMultiTable['tables'],
+  onFilter: (data: LensFilterEvent['data']) => void
+): LegendAction => ({ series: [xySeries], label }) => {
```

Comment:
This is a react component and elastic-charts will re-render it a lot (e.g. when moving the mouse over the chart, changing the hover target will trigger a re-render on each legend item every time).

It seems worth adding a `React.memo(({ series: [...` in there.

Reply:
Fixed! Thanx Joe!